authenticationsdk/util/MLEUtility.py

```diff
@@ -1,21 +1,140 @@
+    def encrypt_request_payload(merchant_config, request_body):
+        
+        if request_body is None or request_body == "":
+            return request_body
+
+        try:
+            cert = MLEUtility.get_certificate(merchant_config)
+        except Exception as e:
+            raise MLEUtility.MLEException(f"Unable to load certificate: {str(e)}")
+
+        try:
+            public_key = cert.public_key()
+            serial_number = MLEUtility.extract_serial_number(cert)
+            if serial_number is None:
+                raise ValueError("Serial number could not be fetched")
+
+            jwk_key = jwk.JWK.from_pyca(public_key)
+            
+            MLEUtility.logger.debug(f"Request payload encrypted successfully. {request_body}")
+
+            payload = request_body.encode('utf-8')
+
+            header = {
+                "alg": "RSA-OAEP-256",
+                "enc": "A256GCM",
+                "cty": "JWT",
+                "kid": serial_number,
+                "iat": int(time.time())
+            }
 
-        # return encrypted requestBody
-        return requestBody
+            jwe_token = jwe.JWE(plaintext=payload,
+                                protected=json.dumps(header))
+
+            jwe_token.add_recipient(jwk_key)
+
+            encrypted_request_body = jwe_token.serialize(compact=True)
+            MLEUtility.logger.debug(f"Request payload encrypted successfully. {encrypted_request_body}")
+            return MLEUtility.create_json_object(encrypted_request_body)
+        except Exception as e:
+            MLEUtility.logger.error(f"Error encrypting request payload: {str(e)}")
+            raise MLEUtility.MLEException(f"Error encrypting request payload: {str(e)}")
+
+    @staticmethod
+    def get_certificate(merchant_config):
+        cache_obj = FileCache()
+        try:
+            cert_data = cache_obj.grab_file(merchant_config, merchant_config.key_file_path,
+                                                merchant_config.key_file_name)
+            certificate = cert_data[3]
+            if certificate is not None:
+                MLEUtility.validate_certificate_expiry(certificate, merchant_config.get_mleKeyAlias())
+                MLEUtility.logger.debug(f"Certificate found for MLE with alias {merchant_config.get_mleKeyAlias()}")
+                return certificate
+            else:
+                MLEUtility.logger.error(
+                    f"No certificate found for MLE for given mleKeyAlias {merchant_config.get_mleKeyAlias()} in p12 file {merchant_config.key_file_name}.p12")
+                raise MLEUtility.MLEException(
+                    f"No certificate found for MLE for given mleKeyAlias {merchant_config.get_mleKeyAlias()} in p12 file {merchant_config.key_file_name}.p12")
+        except KeyError:
+            MLEUtility.logger.error(
+                f"No certificate found matching the MLE key alias: {merchant_config.get_mleKeyAlias()}")
```

Comment:
raise MLEUtility.MLEException(
                    f"No certificate found for MLE for given mleKeyAlias {merchant_config.get_mleKeyAlias()} in p12 file {merchant_config.key_file_name}.p12")